MuxExoPlayer/src/r2_12_1/java/com/mux/stats/sdk/muxstats/MuxStatsExoPlayer.java

```diff
@@ -0,0 +1,379 @@
+
+import java.io.IOException;
+
+public class MuxStatsExoPlayer extends MuxBaseExoPlayer implements AnalyticsListener, Player.EventListener{
+
+    public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+                             CustomerPlayerData customerPlayerData,
+                             CustomerVideoData customerVideoData) {
+        this(ctx, player, playerName, customerPlayerData,
+                customerVideoData, null, true);
+    }
+
+    public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+                             CustomerPlayerData customerPlayerData,
+                             CustomerVideoData customerVideoData,
+                             CustomerViewData customerViewData) {
+        this(ctx, player, playerName, customerPlayerData, customerVideoData,
+                customerViewData, true);
+    }
+
+    public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+                             CustomerPlayerData customerPlayerData,
+                             CustomerVideoData customerVideoData,
+                             boolean sentryEnabled) {
+        this(ctx, player, playerName, customerPlayerData, customerVideoData,
+                null, sentryEnabled);
+    }
+
+    public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+                             CustomerPlayerData customerPlayerData,
+                             CustomerVideoData customerVideoData,
+                             CustomerViewData customerViewData, boolean sentryEnabled) {
+        this(ctx, player, playerName, customerPlayerData, customerVideoData,
+                null, sentryEnabled, new MuxNetworkRequests());
+    }
+
+    public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+                             CustomerPlayerData customerPlayerData,
+                             CustomerVideoData customerVideoData,
+                             CustomerViewData customerViewData, boolean sentryEnabled,
+                             INetworkRequest networkRequests ) {
+        super(ctx, player, playerName, customerPlayerData, customerVideoData,
+                customerViewData, sentryEnabled, networkRequests);
+
+        if (player instanceof SimpleExoPlayer) {
+            ((SimpleExoPlayer) player).addAnalyticsListener(this);
+        } else {
+            player.addListener(this);
+        }
+        if (player.getPlaybackState() == Player.STATE_BUFFERING) {
+            // playback started before muxStats was initialized
+            play();
+            buffering();
+        } else if (player.getPlaybackState() == Player.STATE_READY) {
+            // We have to simulate all the events we expect to see here, even though not ideal
+            play();
+            buffering();
+            playing();
+        }
+    }
+
+    @Override
+    public void release() {
+        if (this.player.get() != null) {
```

Comment:
We need to add the check for `this.player` not being null here, like you did on all of the other versions. I missed this when I made the first changes to add 2.12 support